app/screens/settings/advanced/index.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {View, TouchableOpacity} from 'react-native';
+import {Edge, SafeAreaView} from 'react-native-safe-area-context';
+
+import OptionItem from '@components/option_item';
+import {useTheme} from '@context/theme';
+import useAndroidHardwareBackHandler from '@hooks/android_back_handler';
+import {popTopScreen} from '@screens/navigation';
+import {deleteFileCache, getAllFilesInCachesDirectory, getFormattedFileSize} from '@utils/file';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type {ReadDirItem} from 'react-native-fs';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        screen: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+        },
+        body: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.06),
+            flex: 1,
+            paddingTop: 35,
+        },
+        itemStyle: {
+            backgroundColor: theme.centerChannelBg,
+            paddingHorizontal: 8,
+        },
+    };
+});
+
+const EMPTY_FILES: ReadDirItem[] = [];
+const EMPTY_SERVERS: string[] = [];
+const EDGES: Edge[] = ['left', 'right'];
+
+type AdvancedSettingsProps = {
+    componentId: string;
+}
+const AdvancedSettings = ({componentId}: AdvancedSettingsProps) => {
+    const theme = useTheme();
+    const intl = useIntl();
+    const [dataSize, setDataSize] = useState<number|undefined>(0);
+    const [files, setFiles] = useState<ReadDirItem[]>(EMPTY_FILES);
+    const [serverUrls, setServerUrls] = useState<string[]>(EMPTY_SERVERS);
+    const styles = getStyleSheet(theme);
+
+    const getAllCachedFiles = useCallback(async () => {
+        const {totalSize, files: cachedFiles, serverUrls: allServerUrls} = await getAllFilesInCachesDirectory();
+        setDataSize(totalSize);
+        setFiles(cachedFiles || EMPTY_FILES);
+        setServerUrls(allServerUrls || EMPTY_SERVERS);
+    }, []);
+
+    const onPressDeleteData = preventDoubleTap(async () => {
```

Comment:
The way it is implemented right now, there seems no need to have `getAllCachedFiles` to use `useCallback`. `onPressDeleteData` could use useCallback, and then you could use the memoized `getAllCachedFiles` to simplify the dependencies, but... `getAllCachedFiles` don't depend on anything, so... maybe not needed there at all.

Reply:
refactored